### PR TITLE
boards: x86: depend on CONFIG_BUILD_OUTPUT_EFI

### PR DESCRIPTION
**Pull request by nashif**
Add a new Kconfig CONFIG_BUILD_OUTPUT_EFI and select that for boards
that want to generate an EFI application.
Make qemu_x86_64 also generate an EFI file, however do not enable this
by default yet.

Goal is to boot qemu using EFI to be able to test this path in the
future.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

File: Kconfig.zephyr [hunk 1]
```diff
@@ -317,6 +317,19 @@ config BUILD_OUTPUT_BIN
 	  Build a "raw" binary zephyr/zephyr.bin in the build directory.
 	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
 
+config BUILD_OUTPUT_EFI
+	bool "Build as an EFI application"
+	default n
+	help
```
Comment (galak) [inline Kconfig.zephyr:323]: Should we have this depend on x86_64 right now?  Not sure if we support EFI on any other arch's?

**Review by tyrantlizard77 — APPROVED** [review]
needs rebase but LGTM

File: boards/x86/ehl_crb/ehl_crb_defconfig [hunk 2]
```diff
@@ -10,3 +10,5 @@ CONFIG_UART_NS16550=y
 CONFIG_UART_CONSOLE=y
 CONFIG_X2APIC=y
 CONFIG_SMP=y
+CONFIG_KERNEL_VM_SIZE=0x80800000
```
Comment (jhedberg) [inline boards/x86/ehl_crb/ehl_crb_defconfig:13]: Was this a rebasing mistake? Why was the explicit KERNEL_VM_SIZE brought back?

File: boards/x86/up_squared/up_squared_defconfig [hunk 3]
```diff
@@ -11,3 +11,5 @@ CONFIG_UART_CONSOLE=y
 CONFIG_X2APIC=y
 CONFIG_SMP=y
 CONFIG_MP_NUM_CPUS=2
+CONFIG_KERNEL_VM_SIZE=0x80800000
```
Comment (jhedberg) [inline boards/x86/up_squared/up_squared_defconfig:14]: Same here wrt KERNEL_VM_SIZE

**Pull request merged**